pwndbg/heap/ptmalloc.py

```diff
@@ -32,6 +32,83 @@ def heap_for_ptr(ptr):
     return ptr & ~(HEAP_MAX_SIZE - 1)
 
 
+class Chunk:
+    def __init__(self, addr):
+        self._gdbValue = pwndbg.gdblib.memory.poi(pwndbg.heap.current.malloc_chunk, addr)
+        self.address = int(self._gdbValue.address)
+        self._prev_size = None
+        self._size = None
+        self._flags = None
+        self._fd = None
+        self._bk = None
+        # TODO fd_nextsize, bk_nextsize, key, REVEAL_PTR etc.
+
+    @property
+    def prev_size(self):
+        if self._prev_size is None:
+            if "mchunk_prev_size" in (field.name for field in self._gdbValue.type.fields()):
+                prev_size_field_name = "mchunk_prev_size"
+            else:
+                prev_size_field_name = "prev_size"
+
+            try:
+                self._prev_size = int(self._gdbValue[prev_size_field_name])
+            except gdb.MemoryError:
+                pass
+
+        return self._prev_size
+
+    @property
+    def size(self):
+        if self._size is None:
+            if "mchunk_size" in (field.name for field in self._gdbValue.type.fields()):
+                size_field_name = "mchunk_size"
+            else:
+                size_field_name = "size"
+
+            try:
+                self._size = int(self._gdbValue[size_field_name])
+            except gdb.MemoryError:
+                pass
+
+        return self._size
+
+    @property
+    def flags(self):
+        if self._flags is None:
+            sz = self.size
+            if sz is not None:
+                self._flags = {
+                    "non_main_arena": bool(sz & ptmalloc.NON_MAIN_ARENA),
+                    "is_mmapped": bool(sz & ptmalloc.IS_MMAPPED),
+                    "prev_inuse": bool(sz & ptmalloc.PREV_INUSE),
+                }
+
+        return self._flags
+
+    @property
+    def fd(self):
```

Comment:
I feel like there should be a way to simplify this "lazy" pattern we're using through out these functions, so the code looks simpler and we can reuse it in other places we want lookups to be lazy:
```
if self.FIELD is None:
    try:
        self.FIELD = int(self._gdbvalue[FIELD])
    except gdb.MemoryError:
        pass
return self.FIELD
```

But I don't know the best way to do that right now. Not a big deal for this PR, just something to think about.

Reply:
Yes this could be challenging, since not all properties follow the same pattern e.g. `flags` require the chunk's `size` field to be fetched first. I'll leave this for now.